docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,229 @@
+   - If the input is `Yes`, it will generate x amount of invitation tokens per user based on the input from CLI and save them into `tokens` table.
+   - The CLI will call `ClearUsers` endpoint to set `eligibleUser` to be empty if the input is `No`.
+
+_Referral Link Redemption_
+
+1. User Alice tries to register a new account through a referral link, which triggers redeemed satellite to verify invitation token using Referral manager.
+2. Referral Manager checks the status of the token:
+    - if the token exists in `tokens` table:
+        - if it is not redeemed, Referral Manager sends back a success response to the redeemed satellite and mark the token as redeemed in the Referral Manager's database
+        - if it token is already redeemed, Referral Manager sends back a `invalid token` response to the redeemed satellite.
+        
+    - if the token doesn't exist in `tokens` table, Referral Manager sends back a `invalid token` response to the redeemed satellite.
+3. Redeemed satellite receives the response, which:
+    - if it is a success, the satellite will proceed with the account creation and
+    then send a request to the Referral Manager, which: 
+        - updates the `tokens` table to store the newly created user ID in `redeemedID` column for that particular referral token.
+        - updates the `users` table to store the newly created user ID and the satellite URL the user belongs to.
+    - if it is an invalid token, the satellite will display a proper message in the UI.
+    
+## Rationale
+
+We will create a `inspector` port that's only listening on `localhost` so CLI could be only used by Referral Manager operators.
+
+We could also set the `inspector` port to only accept requests coming from storj vpn.
+
+## Implementation
+
+- Create a private repository for Referral Manager.
+- Create `tokens` and `users` table in Referral Manager database.
+- Create an `inspector` endpoint on Referral Manager.
+    - Implementing a method `GenerateTokens` for generating referral tokens and save them into `tokens` table on Referral Manager.
+    - Implementing a mathod `GetEligibleUsers` for querying users who don't have any referral tokens associated with.
+- Implementing an endpoint `GetTokens` for requesting unredeemed tokens from Referral Manager. 
+- Implementing an endpoint `Redeem` on Referral Manager for verifying invitation tokens.
+- Implementing an endpoint `ClearUsers` on Referral Manager for setting in-memory `eligibleUsers` to be empty from Referral Manager CLI.
+- Replace existing registration token logic.
+
+### Pseudocode
+
+The Token struct represents both a Go struct (on satellite and Referral Manager) as well as the schema for the `tokens` table on the Referral Manager
+
+```
+type Token struct {
+	Secret [32]byte
+	OwnerID uuid.UUID
+	RedeemedID uuid.UUID
+	HostSatelliteURL string
+    RedeemedSatelliteURL string
```

Comment:
the difference in indentation here looks like an easy mistake, but just to make sure- should this be a tab instead of 4 spaces?